File: packages/dotnet/Pezhwan/SessionService.cs
namespace Pezhwan;

/// <summary>
/// Session endpoints: list, revoke one, revoke all.
/// </summary>
public sealed class SessionService
{
    private readonly PezhwanClient _client;

    internal SessionService(PezhwanClient client)
    {
        _client = client;
    }

    /// <summary>Lists the authenticated user's active sessions.</summary>
    public async Task<List<Models.Session>> ListAsync(CancellationToken ct = default)
    {
        var data = await _client.GetAsync("/v1/sessions", ct).ConfigureAwait(false);
        if (!data.TryGetValue("sessions", out var raw) || raw is not System.Collections.IEnumerable seq)
            return new List<Models.Session>();
        var result = new List<Models.Session>();
        foreach (var item in seq)
        {
            if (item is Dictionary<string, object?> map)
                result.Add(Models.Session.From(map));
        }
        return result;
    }

    /// <summary>Revokes a single session.</summary>
    public Task RevokeAsync(string sessionId, CancellationToken ct = default)
        => _client.DeleteAsync($"/v1/sessions/{sessionId}/revoke", ct);

    /// <summary>Revokes every session for the current user.</summary>
    public Task RevokeAllAsync(CancellationToken ct = default)
        => _client.PostAsync("/v1/sessions/all/revoke", new Dictionary<string, object?>(), ct);
}

File: packages/java/src/main/java/com/pezhwan/SessionService.java
package com.pezhwan;

import java.io.IOException;
import java.util.List;
import java.util.Map;

/**
 * Session endpoints: list, revoke one, revoke all.
 */
public final class SessionService {

    private final PezhwanClient client;

    SessionService(PezhwanClient client) {
        this.client = client;
    }

    /** Lists the authenticated user's active sessions. */
    public List<Models.Session> list() throws IOException, InterruptedException {
        Map<String, Object> data = client.get("/v1/sessions");
        Object raw = data.get("sessions");
        if (!(raw instanceof List<?> list)) {
            return List.of();
        }
        return list.stream()
                .filter(Map.class::isInstance)
                .map(item -> Models.Session.from(Models.castMap(item)))
                .collect(java.util.stream.Collectors.toList());
    }

    /** Revokes a single session. */
    public void revoke(String sessionId) throws IOException, InterruptedException {
        client.delete("/v1/sessions/" + sessionId + "/revoke");
    }

    /** Revokes every session for the current user. */
    public void revokeAll() throws IOException, InterruptedException {
        client.post("/v1/sessions/all/revoke", Map.of());
    }
}